contracts/pcv/compound/ERC4626CErc20Wrapper.sol

```diff
@@ -0,0 +1,240 @@
+    /// the Vault for accounting, depositing, and withdrawing.
+    IERC20 public immutable cTokenUnderlying;
+
+    /*///////////////////////////////////////////////////////////////
+                Constructor
+    //////////////////////////////////////////////////////////////*/
+    /// @notice ERC4626CErc20Wrapper constructor
+    /// @param _cToken Compound cToken to wrap
+    /// @param name ERC20 name of the vault shares token
+    /// @param symbol ERC20 symbol of the vault shares token
+    constructor(
+        address _cToken,
+        string memory name,
+        string memory symbol
+    ) ERC20(name, symbol) {
+        require(CErc20(_cToken).isCToken(), "ERC4626Cerc20Wrapper: not a cToken");
+        require(!CErc20(_cToken).isCEther(), "ERC4626Cerc20Wrapper: cEther not supported");
+        cToken = CErc20(_cToken);
+        cTokenUnderlying = IERC20(CErc20(cToken).underlying());
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                ERC4626 Vault properties
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice The address of the underlying ERC20 token used for
+    /// the Vault for accounting, depositing, and withdrawing.
+    function asset() external view returns(address) {
+        return address(cTokenUnderlying);
+    }
+
+    /// @notice Total amount of the underlying asset that
+    /// is "managed" by Vault.
+    function totalAssets() external view returns(uint256) {
+        return cToken.getCash() + cToken.totalBorrows();
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                ERC4626 Deposit/Withdrawal logic
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Mints `shares` Vault shares to `receiver` by
+    /// depositing exactly `amount` of underlying tokens.
+    function deposit(uint256 assets, address receiver) external returns(uint256 shares) {
+        // transfer user tokens to self
+        cTokenUnderlying.safeTransferFrom(msg.sender, address(this), assets);
+
+        // mint cTokens
+        cTokenUnderlying.approve(address(cToken), assets);
+        uint256 balanceBefore = cToken.balanceOf(address(this));
+        require(cToken.mint(assets) == 0, "ERC4626CErc20Wrapper: error on cToken.mint");
+        shares = cToken.balanceOf(address(this)) - balanceBefore;
+
+        // mint shares to receiver
+        _mint(receiver, shares);
+
+        // emit event
+        emit Deposit(msg.sender, receiver, assets, shares);
+
+        return shares;
+    }
+
+    /// @notice Mints exactly `shares` Vault shares to `receiver`
+    /// by depositing `amount` of underlying tokens.
+    function mint(uint256 shares, address receiver) external returns(uint256 assets) {
+        // compute exact amount of assets to send
+        uint256 exchangeRate = cToken.exchangeRateCurrent();
+        assets = shares * exchangeRate / 1e18;
+
+        // mint cTokens
```

Comment:
Can probably share a lot of the underlying logic in a deposit function

Reply:
factorized the code